--- magic_overlayfs/magisk-module/customize.sh	2024-08-01 15:49:54
+++ magic_overlayfs/magisk-module/customize.sh	2024-08-01 11:57:52
@@ -68,6 +68,8 @@
                      mode.sh \
                      mount.sh \
                      uninstall.sh \
+                     sepolicy.rule \
+                     system.prop \
                      module.prop \
                      "libs/$ABI/overlayfs_system" \
                      -d "$MODPATH"
