--- magisk-module/module.prop	2024-07-31 17:12:34
+++ magisk-module/module.prop	2024-08-01 10:48:34
@@ -1,7 +1,6 @@
-id=magisk_overlayfs
-name=Magical Overlayfs
-author=HuskyDG
-version=v3.2.2
-versionCode=30202
-description=Use OverlayFS to make read-only system partitions become read-write partitions. OverlayFS MUST be supported by kernel!
-updateJson=https://github.com/huskydg/magic_overlayfs/releases/latest/download/update_release.json
+id=magisk_overlayfs+adb-root-enabler
+name=Magical Overlayfs + Adb-Root-Enabler
+author=HuskyDG + anasfanani, merged by agreenbhm
+version=v3.2.2+v1.2
+versionCode=1
+description=Use OverlayFS to make read-only system partitions become read-write partitions. OverlayFS MUST be supported by kernel! + Enable ADB root & authenticate
